### PR TITLE
Add AT_CMD_PROMPT as valid state for at_cmd_write()

### PR DESCRIPTION
**Pull request by brendanyounger**
Allow at_cmd_write() to recognize ">" as a valid return value with state AT_CMD_PROMPT rather than treating it like a notification.  AT+CMGS will return a ">" prompt when sending the command as two parts.

### CODE REVIEW

File: drivers/at_cmd/at_cmd.c [hunk 1]
```diff
@@ -143,13 +151,13 @@ static void socket_thread_fn(void *arg1, void *arg2, void *arg3)
 		item->callback = NULL;
 
 		bytes_read = recv(common_socket_fd, item->data,
-				  sizeof(item->data), 0);
+					sizeof(item->data), 0);
```
Comment (rlubos) [inline drivers/at_cmd/at_cmd.c]: Please don't modify proper code format. Here and in a few places below.

**Comment by evenl** [comment]
I have never seen '>' in any AT responses, and I am not able to find it in any specification. Can you list any document or specification that describe the use of '>' in AT command responses? 

Can you give some more information about what you are trying to do? Maybe with some examples where you need the at_cmd_write function to recognize the '>' character?

**Comment by brendanyounger** [comment]
`at_cmd_write("AT+CMGS=2\r", NULL, 0, NULL)` or just AT+CMGS=2 at the at_host command line will return a ">" prompt for the PDU of the SMS to send.  Right now, ">" is parsed as a URC like +CME ERROR when it's really the equivalent of "OK".

File: drivers/at_cmd/at_cmd.c (hunk 1 above)
Comment (brendanyounger) [reply drivers/at_cmd/at_cmd.c]: Those were added automatically by my editor.  Do you have a reference for the appropriate style I can use with Sublime Text?  What editor do the Nordic folks use?

Comment (rlubos) [reply drivers/at_cmd/at_cmd.c]: We generally follow [Zephyr coding style guidelines](https://docs.zephyrproject.org/latest/contribute/index.html#coding-style) here, which in turn follows Linux kernel style. 

We have a `clang-format` config file in this repository, so you can run `git-clang-format` after you stage your changes to see if they comply. Keep in mind though, that this tool has some issues, unfortunately - for instance, I've experienced problems with structure formatting several times, so I needed to fix them manually.

I'm not familiar with Sublime, so I'm not able to comment if there're any format configs for it.
Nordic officially recommended IDE is Segger Embedded Studio. Personally, I use VS code,, have no issues with it.

**Comment by evenl** [comment]
In your example you don't have any return string buffer, so the at command driver is not able to return anything to you. I don't think it's the at cmd driver or the modem that is returning the > character you are seeing. I think it may be your terminal application. 

I did a slight modification to your code, 

`err = at_cmd_write("AT+CMGS=2\r", str, 256, NULL);`

I am not able to replicate what you are seeing, the only thing I get back is an empty string and the error code -8. Which is to be expected in this case as the AT command string is malformed.

**Comment by evenl** [comment]
I got some more information about this feature now, so this is what is called "interactive mode". This did not exist when I wrote the first itteration of this driver, and is an convenience feature to easier send SMS from an terminal. 

This mode should not be supported by the at_cmd driver as it would then need to hold back all writes from other threads until the thread that enabled interactive mode is done. From an application perspective this mode does not make sense as the application should be able create the correct at+cmgs string without this mode. 

There is an SMS module in the works for NCS, currently it only supports reception of SMSes, but and feature request could be added to make it support sending SMSes to. 
https://github.com/NordicPlayground/fw-nrfconnect-nrf/pull/1633/files
I think that a better solution to this issue than supporting interactive mode in the at_cmd driver.

**Comment by carlescufi** [comment]
Note: This is an automated message.
@brendanyounger this Pull Request has not been updated for more than a month, please close it if not applicable anymore.
@metc  @anangl  @pizi-nordic  please close it if the author is not available

**Comment by rlubos** [comment]
Stale

**Pull request closed without merging**